lib/repository.py

```diff
@@ -41,6 +42,13 @@ def get_git_repository(name, remote_repo_url, parent_dir_path):
     dir, according to specified name.
     If it does not exist, clone it from the specified URL.
     """
+    # infer name if it is not provided
+    if not name:
+        match_obj = re.match(".*/([a-zA-Z0-9\-]*)\.git", remote_repo_url)
```

Comment:
Won't this break if the repository name is some weird string?

There are some [special characters that are allowed in a URL](http://stackoverflow.com/questions/1856785/characters-allowed-in-a-url#1856809) that could show up in the repository's name.